packages/replay/src/coreHandlers/handleScope.ts

```diff
@@ -26,6 +26,10 @@ export const handleScopeListener: (replay: ReplayContainer) => (scope: Scope) =>
  * An event handler to handle scope changes.
  */
 export function handleScope(scope: Scope): Breadcrumb | null {
+  if (typeof scope.getLastBreadcrumb !== 'function') {
```

Comment:
shouldn't this be `scope?.getLastBreadcrumb` as it would break on this condition too, before checking the `typeof`

otherwise it's LGTM. 👍🏼 

I wish i would have created this PR before 🤦🏼 🥼

Reply:
According to this methods signature, `scope` is always defined; so no need to guard against it being undefined ;)
Furthermore, this is where we call the function that ends up calling `handleScope`. As you can see, it is guarded by the scope being defined there: https://github.com/getsentry/sentry-javascript/blob/a6a58e394f210dff6550af64aaa4bd93e5ffea49/packages/replay/src/util/addGlobalListeners.ts#L19

Reply:
awesome sauce... 🆒 .   The guarded scope looks good ⚡

Reply:
l: I would just do `if (scope.getLastBreadcrumb)`, saves some bytes and should be good enough here!

Reply:
[2690545](https://github.com/getsentry/sentry-javascript/pull/6969/commits/2690545ebcee2d80b398d22359d51e67d5acb69f)